File: ch06/user-registration/course-tracker-final/src/main/java/com/manning/sbip/ch06/model/Course.java
package com.manning.sbip.ch06.model;

import javax.persistence.*;
import javax.validation.constraints.*;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "COURSES")
@NoArgsConstructor
public class Course {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "ID")
	private long id;

	@Column(name = "NAME")
	@NotEmpty(message = "Course name field can't be empty")
	private String name;

	@Column(name = "CATEGORY")
	@NotEmpty(message = "Course category field can't be empty")
	private String category;

	@Column(name = "RATING")
	@Min(value = 1)
	@Max(value = 5)
	private int rating;

	@Column(name = "DESCRIPTION")
	@NotEmpty(message = "Course description field can't be empty")
	private String description;
}

File: ch10/Kotlin/course-tracker-kotlin-app-security/src/main/kotlin/com/manning/sbip/ch10/model/Course.kt
package com.manning.sbip.ch10.model

import javax.persistence.*;
import javax.validation.constraints.*;

@Entity
@Table(name = "Courses")
data class Course(

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    @field:Column(name = "ID")
    var id: Long? = 0,

    @field:Column(name = "NAME")
    @field:NotEmpty(message = "Course name field can't be empty")
    var name: String? = "",

    @field:Column(name = "CATEGORY")
    @field:NotEmpty(message = "Course category field can't be empty")
    var category: String? = "",

    @field:Column(name = "RATING")
    @Min(value = 1)
    @Max(value = 5)
    var rating : Int? = 0,

    @field:Column(name = "DESCRIPTION")
    @field:NotEmpty(message = "Course description field can't be empty")
    var description: String? = ""
)
